dom/fuzzer/modules/web-speech.js

```diff
@@ -0,0 +1,173 @@
+          }],
+/*
+      [1, function(){
+            // create a recognition
+            var myrecognition = Things.reserve();
+            cmd.push(myrecognition + ' = new SpeechRecognition();');
+            recognitions.push(myrecognition);
+          }],
+      [1, function(){
+            // set attribute on a recognition
+            if (!recognitions.length) return;
+            cmd.push(JS.setAttribute(Random.pick(recognitions), recognitionAttributes) + ';');
+          }],
+      [1, function(){
+            // call something on SpeechRecognition
+            if (!recognitions.length) return;
+            cmd.push(JS.methodCall(Random.pick(recognitions), recognitionMethods) + ';');
+          }],
+      [1, function(){
+            // set a recognition event
+            if (!recognitions.length) return;
+            cmd.push(Random.pick(recognitions) + ".addEventListener" + JS.methodHead([simpleSource(Random.pick(recognitionEvents)), "function(e) { " + fuzzSubCommand() + "}"]) + ';');
+          }],
+      [1, function(){
+            // create a grammar
+            var mygrammar = Things.reserve();
+            cmd.push(mygrammar + ' = new SpeechGrammar();');
+            grammars.push(mygrammar);
+          }],
+      [1, function(){
+            // set attribute on a grammar
+            var g = getGrammar();
+            if (g === undefined) return;
+            cmd.push(JS.setAttribute(g, grammarAttributes) + ';');
+          }],
+      [1, function(){
+            // create a grammar list
+            var mygrammarl = Things.reserve();
+            cmd.push(mygrammarl + ' = new SpeechGrammarList();');
+            grammarLists.push(mygrammarl);
+          }],
+      [1, function(){
+            // call something on SpeechGrammarList
+            if (!grammarLists.length) return;
+            cmd.push(JS.methodCall(Random.pick(grammarLists), grammarListMethods) + ';');
+          }],
+*/
+    ]);
+    return cmd;
+  };
+
+  var getGrammar = function(){
+    var poss = [];
+    if (grammarLists.length)
+      poss.push([10, function(){ var gl = Random.pick(grammarLists); return gl + '[' + rnd(1024) + '%' + gl + '.length]'; }]);
+    if (grammars.length)
+      poss.push([1, grammars]);
+    if (!poss.length)
+      return undefined;
+    return Random.choose(poss);
+  };
+
+  var utteranceAttributes = {
+    'text': function(){ return simpleSource((rnd(10) === 0) ? '' : Random.pick(thingsToSay)); },
+    'lang': function(){ return simpleSource(Random.pick(fuzzValues.languages)); },
+    'voiceURI': function() { if (rnd(32)===0) return simpleSource(Random.pick(fuzzValues.URIs)); return '(function(){var vl=window.speechSynthesis.getVoices();return vl['+rnd(1024)+'%vl.length].voiceURI;})()'; }, // XXX: get from window.speechSynthesis.getVoices() at runtime
```

Comment:
If you're worried about the side effects from calling `window.speechSynthesis.getVoices();` within makeCommand, you can emit code that calls it and sets `fuzzerWebSpeech.nextVoice` for the next call to makeCommand to use. See WebIDL.js and reflection-and-torture.js for complete examples.